MiniSim/MiniSim.swift

```diff
@@ -14,9 +14,12 @@ import UserNotifications
 class MiniSim: NSObject {
     private var menu: Menu!
     
-    @objc let statusItem = NSStatusBar.system.statusItem(withLength: NSStatusItem.variableLength)
+    @objc var statusItem = NSStatusBar.system.statusItem(withLength: NSStatusItem.variableLength)
     
     private var isOnboardingFinishedObserver: NSKeyValueObservation?
+    private var menuImageObserver: NSKeyValueObservation?
+    
+    private var menuImagesObserved: [String] = []
```

Comment:
Why do we need to store those images? This looks unnecessary

Reply:
menuImagesObserved is used because without a conditional that uses it (line 121), when you select an icon again that was already used before, hence already resized and padded, it'll get smaller and smaller. This prevents this from happening. There could be a cleaner approach to this, but I'm not sure.